src/rp2/plugin/accounting_method/hifo.py

```diff
@@ -51,14 +52,20 @@ def seek_non_exhausted_acquired_lot(
                 # The acquired lot has zero partial amount
                 continue
 
-            if selected_acquired_lot is None or selected_acquired_lot.spot_price < acquired_lot.spot_price:
+            if acquired_lot_amount > ZERO:
                 selected_acquired_lot_amount = acquired_lot_amount
                 selected_acquired_lot = acquired_lot
+                break
 
         if selected_acquired_lot_amount > ZERO and selected_acquired_lot:
             lot_candidates.clear_partial_amount(selected_acquired_lot)
+            if selected_acquired_lot_amount > taxable_event_amount:
```

Comment:
Could you clarify this bit (it's the same as in lifo)?

Reply:
Same as below - if there's any leftover amount in the lot add it back to the heap.

Reply:
Oh, I see, it's because you popped it earlier, so you need to push it back on the heap: this bit is necessary because it's a heap-based class (I was still thinking in terms of FIFO). Makes sense.